examples/defaulter-gen/generators/defaulter.go

```diff
@@ -387,6 +391,25 @@ func Packages(context *generator.Context, arguments *args.GeneratorArgs) generat
 	return packages
 }
 
+// callTreeForType contains fields necessary to call buildCallTreeForType.
+type callTreeForType struct {
+	t                      *types.Type
+	root                   bool
+	existingDefaulters     defaulterFuncMap
+	newDefaulters          defaulterFuncMap
+	currentlyBuildingTypes map[*types.Type]bool
+}
+
+func NewCallTreeForType(t *types.Type, root bool, existingDefaulters, newDefaulters defaulterFuncMap, currentlyBuildingTypes map[*types.Type]bool) callTreeForType {
```

Comment:
We can't initialize currentlyBuildingTypes in newCallTreeForType because when this is encountered:

```
r := newCallTreeForType(c.existingDefaulters, c.newDefaulters)
	if child := r.build(t.Elem, false); child != nil
```

it will create a new map for currentlyBuildingTypes - destroying our purpose for storing the types that are currently building. :)